packages/replayio/src/utils/async/logPromise.ts

```diff
@@ -2,30 +2,29 @@ import { dots } from "cli-spinners";
 import { isDebugging } from "../../config";
 import { logUpdate } from "../logUpdate";
 import { statusFailed, statusPending, statusSuccess } from "../theme";
-import { STATUS_PENDING, STATUS_REJECTED, STATUS_RESOLVED, Status } from "./createDeferred";
+import { STATUS_PENDING, STATUS_REJECTED, STATUS_RESOLVED, createDeferred } from "./createDeferred";
 
-export async function logPromise(
-  promise: Promise<any>,
+export async function logPromise<T>(
```

Comment:
tiny nit: I prefer descriptive names over single letter variable or generic type names. Single letter names are sometimes (not always) vague to people who didn't write the code originally, and I prefer clarity/consistency to avoid those cases.
```suggestion
export async function logPromise<PromiseType>(
```